### PR TITLE
api: http: make consistent api and implementation

### PR DESCRIPTION
**Pull request by jcvenegas**
cpu: cpu is an number under the implementation not struct, unless we
need to add more values to the cpu it make sense keep it as integer.

vsocks: vsocks is implemented as an array

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: vmm/src/api/openapi/cloud-hypervisor.yaml [hunk 1]
```diff
@@ -213,23 +215,15 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/VhostUserBlkConfig'
-        vsock:
-          $ref: '#/components/schemas/VsockConfig'
+        vsocksss:
```
Comment (sameo) [inline vmm/src/api/openapi/cloud-hypervisor.yaml]: Two too many `s` ?

File: vmm/src/api/openapi/cloud-hypervisor.yaml [hunk 2]
```diff
@@ -172,7 +172,9 @@ components:
       type: object
       properties:
         cpus:
-          $ref: '#/components/schemas/CpuConfig'
+          minimum: 1
+          default: 1
+          type: integer
```
Comment (sameo) [inline vmm/src/api/openapi/cloud-hypervisor.yaml]: I'd prefer if we keep a structure for `CpuConfig`. We may extend it to include more information in the future.

File: vmm/src/api/openapi/cloud-hypervisor.yaml [hunk 3]
```diff
@@ -213,8 +213,10 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/VhostUserBlkConfig'
-        vsock:
-          $ref: '#/components/schemas/VsockConfig'
+        vsocks:
```
Comment (sboeuf) [inline vmm/src/api/openapi/cloud-hypervisor.yaml]: why did you change the name `vsock` into `vsocks`?

File: vmm/src/config.rs [hunk 4]
```diff
@@ -131,25 +131,43 @@ fn parse_iommu(iommu: &str) -> Result<bool> {
...
-        Ok(CpusConfig(
-            cpus.parse::<u8>().map_err(Error::ParseCpusParams)?,
-        ))
+        // Split the parameters based on the comma delimiter
+        let params_list: Vec<&str> = cpus.split(',').collect();
+
+        let mut cpu_count_str: &str = "";
+
+        for param in params_list.iter() {
+            if param.starts_with("cpu_count=") {
```
Comment (sboeuf) [inline vmm/src/config.rs]: Here you're modifying the expectation about the CLI too. This needs further changes to the documentation if we accept this change.

But I'm wondering why you couldn't leave the CpusConfig structure the way it was? Was the deserialization failing?

Comment (jcvenegas) [reply vmm/src/config.rs]: @sboeuf yeah , was a deserialization  error when http request tried to get information from vmconfig.
As defined in the api yaml, I am sending :
`{"cpus":{"cpu_count":5},"memory":{"size":0},"kernel":{"path":""},"cmdline":{"args":"hola"},"rng":{"src":""},"serial":{"mode":"Off"},"console":{"mode":"Off"}}`

but it fails saying `SerdeJsonDeserialize(Error("invalid type: map, expected u8", line: 1, column: 8))` .

File: vmm/src/api/openapi/cloud-hypervisor.yaml (hunk 3 above)
Comment (jcvenegas) [reply vmm/src/api/openapi/cloud-hypervisor.yaml]: it is an array I felt more natural on put in plural.

**Comment by sboeuf** [comment]
@sameo @rbradford PTAL
